src/drivers/sifive_fe310-g000_pll.c

```diff
@@ -297,6 +304,40 @@ static void configure_pll(__metal_io_u32 *pllcfg, __metal_io_u32 *plloutdiv,
 
     __METAL_ACCESS_ONCE(pllcfg) &= ~(PLL_BYPASS);
 
+    /* The PLL needs 100 usec to stabilize before we test PLL_LOCK. Since LFROSC
+     * on all targets with the FE310-G000 PLL runs at 32768 Hz, we need to wait
+     * at least
+     *
+     *   ceil(32768 ticks * 100 usec / 1000000 usec) = 4 ticks
+     *
+     * of mtime before we test PLL_LOCK.
+     *
+     * TODO: Determine the mtime timebase at compile or runtime and use that
+     * here.
+     */
+#ifdef __METAL_DT_RISCV_CLINT0_HANDLE
+    unsigned long long mtime, mtime_end;
+    __metal_driver_riscv_clint0_command_request(__METAL_DT_RISCV_CLINT0_HANDLE,
+                                                METAL_TIMER_MTIME_GET, &mtime);
+    mtime_end = mtime + 4;
+    while (mtime_end > mtime) {
```

Comment:
I think you're off by one here -- if we read the first time near the end of the tick, then we may only wait for 3+ε ticks instead of the required 4.

Something like while (mtime <= mtime_end) should do the trick

Reply:
You're right, thanks!